File: ShoppingListAPI/ShoppingListAPI/Services/ShoppingListService.cs
using ShoppingListAPI.Models;
using ShoppingListAPI.Repositories;

namespace ShoppingListAPI.Services
{
    public class ShoppingListService : IShoppingListService
    {
        private readonly IShoppingListRepository _shoppingListRepository;

        public ShoppingListService(IShoppingListRepository shoppingListRepository)
        {
            _shoppingListRepository = shoppingListRepository;
        }

        public async Task<List<ShoppingItem>> GetShoppingList()
        {
            return await _shoppingListRepository.GetShoppingList();
        }

        public async Task AddItem(ShoppingItem shoppingItem)
        {
            await _shoppingListRepository.AddItem(shoppingItem);
        }

        public async Task RemoveItem(ShoppingItem shoppingItem)
        {
            await _shoppingListRepository.RemoveItem(shoppingItem);
        }
    }
}

File: ShoppingListApp/src/app/service/shopping-list.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ShoppingItem } from '../models/shopping-item';
import { Observable } from 'rxjs/internal/Observable';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ShoppingListService {
  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) { }

  public getShoppingList() : Observable<ShoppingItem[]> {
    return this.http.get<ShoppingItem[]>(this.apiUrl + '/api/shoppinglist/getShoppingList');
  }

  public addShoppingItem(shoppingItem: ShoppingItem) :Observable<ShoppingItem[]> {
    return this.http.post<ShoppingItem[]>(this.apiUrl + '/api/shoppinglist/addItem', shoppingItem);
  }

  public deleteShoppingItem(shoppingItem: ShoppingItem) : Observable<ShoppingItem[]> {
    return this.http.post<ShoppingItem[]>(this.apiUrl + '/api/shoppinglist/removeItem', shoppingItem);
  }
}
